canal/canal.go

```diff
@@ -266,10 +273,15 @@ func (c *Canal) prepareSyncer() error {
 		Port:     uint16(port),
 		User:     c.cfg.User,
 		Password: c.cfg.Password,
+		Charset : c.cfg.Charset,
 	}
 
 	c.syncer = replication.NewBinlogSyncer(&cfg)
 
+        //if len(c.cfg.Charset) != 0 {
```

Comment:
yeah, I have already set character at the replication/binlogsyncer.go, these code is no need